presto-main/src/main/java/com/facebook/presto/execution/resourceGroups/InternalResourceGroup.java

```diff
@@ -555,7 +555,7 @@ public void setSchedulingPolicy(SchedulingPolicy policy)
             eligibleSubGroups = queue;
             while (!queuedQueries.isEmpty()) {
                 ManagedQueryExecution query = queuedQueries.poll();
-                queryQueue.addOrUpdate(query, getQueryPriority(query.getSession()));
+                queryQueue.addOrUpdate(query, getQueryPriority(query));
```

Comment:
Note that this will not affect the priority if the resource group uses `FAIR` scheduling (which is the default).

Reply:
Right, I'm aware of it. Still thinking how we can make a generic interface for "fast-track" queries